packages/core/src/endowmentsToolkit.js

```diff
@@ -172,13 +168,65 @@ function endowmentsToolkit({
   }
 
   /**
-   * @param {string} visitedPath
-   * @param {string[]} pathParts
-   * @param {string[]} explicitlyBanned
-   * @param {object} sourceRef
-   * @param {object} targetRef
-   * @param {object} unwrapTo
-   * @param {object} unwrapFrom
+   * @template {object} T
+   * @param {T} moduleNamespace
+   * @param {string} moduleId
+   * @param {LMPolicy.BuiltinPolicy} policyBuiltin
+   * @returns {Partial<T>}
+   */
+  function getBuiltinForConfig(moduleNamespace, moduleId, policyBuiltin) {
+    /** @type {string[]} */
+    const builtinPaths = []
+
+    /** @type {string[]} */
+    const explicitlyBanned = []
+
+    // Collect the same paths information as getEndowmentsForConfig to enable
+    // matching behavior of policy between globals and builtins
+    Object.entries(policyBuiltin).forEach(([packagePath, allowed]) => {
+      const packagePathParts = packagePath.split('.')
+      if (moduleId === packagePathParts[0]) {
+        const packagePathWithoutPackage = packagePathParts.slice(1).join('.')
+        if (allowed === true) {
+          builtinPaths.push(packagePathWithoutPackage)
+        } else if (allowed === false) {
+          explicitlyBanned.push(packagePathWithoutPackage)
+        }
+      }
+    })
+    const moduleNamespaceView = makeMinimalViewOfRef(
+      moduleNamespace,
+      builtinPaths.sort(),
+      undefined,
+      undefined,
+      explicitlyBanned
```

Comment:
Why is `builtinPaths` sorted, but `explicitlyBanned` is not?

Reply:
AFAIR builtinPaths needs to be sorted for the depth to resolve correctly when iterating over it. explicitlyBanned is only for lookup.